core/trino-main/src/main/java/io/trino/connector/system/TableCommentSystemTable.java

```diff
@@ -159,13 +157,27 @@ private RelationComment getRelationComment(Session session, QualifiedObjectName
             return new RelationComment(true, view.get().getComment());
         }
 
-        RedirectionAwareTableHandle redirectionAware = metadata.getRedirectionAwareTableHandle(session, relationName);
-        if (redirectionAware.tableHandle().isPresent()) {
-            // TODO (https://github.com/trinodb/trino/issues/18514) this should consult accessControl on redirected name. Leaving for now as-is.
-            return new RelationComment(true, metadata.getTableMetadata(session, redirectionAware.tableHandle().get()).getMetadata().getComment());
+        return getCommentOnRedirectedTable(session, relationName);
+    }
+
+    private RelationComment getCommentOnRedirectedTable(Session session, QualifiedObjectName relationName)
+    {
+        String catalog = relationName.getCatalogName();
+        SchemaTableName name = relationName.asSchemaTableName();
+        RedirectionAwareTableHandle redirectedTable = metadata.getRedirectionAwareTableHandle(
+                session,
+                new QualifiedObjectName(catalog, name.getSchemaName(), name.getTableName()));
+
+        if (redirectedTable.redirectedTableName().isEmpty() || redirectedTable.tableHandle().isEmpty()) {
+            return new RelationComment(false, Optional.empty());
+        }
+
+        SchemaTableName redirectedTableName = redirectedTable.redirectedTableName().get().asSchemaTableName();
+        if (!accessControl.filterTables(session.toSecurityContext(), catalog, ImmutableSet.of(redirectedTableName)).contains(redirectedTableName)) {
```

Comment:
the redirected table is (typically) in a different catalog, so `catalog` doesn't seem right here